module/data/item/templates/activities.mjs

```diff
@@ -134,4 +204,133 @@ export default class ActivitiesTemplate extends SystemDataModel {
+          threshold: activity?.attack?.critical?.threshold ?? null,
+          damage: activity?.damage?.critical?.bonus ?? ""
+        };
+      },
+      dammage: () => ({ parts: [], versatile: "" }),
+      enchantment: () => this.activities.getByType("enchant")[0],
+      formula: () => this.activities.getByType("utility")[0]?.roll?.formula ?? "",
+      hasAbilityCheck: () => false,
+      hasAttack: () => !!this.activities.getByType("attack").size,
+      hasDamage: () => !!this.activities.find(a => a.damage?.parts?.length),
+      hasSave: () => !!this.activities.getByType("save").size,
+      hasSummoning: () => !!this.activities.getByType("summon").size,
+      isEnchantment: () => !!this.activities.getByType("enchant").size,
+      isHealing: () => !!this.activities.getByType("heal").size,
+      isVersatile: () => this.actionType && this.properties?.has("ver"),
+      save: () => {
+        const activity = this.activities.getByType("save")[0] ?? {};
+        return {
+          ability: activity.ability ?? null,
+          dc: activity.save?.dc?.formula ?? null,
+          scaling: activity.save?.dc?.calculation ?? ""
+        };
+      },
+      summons: () => this.activities.getByType("summon")[0]
+    });
+
+    addShims("ActivatedEffectTemplate", {
+      activatedEffectCardProperties: () => [
+        this.parent.labels.activation,
+        this.parent.labels.target,
+        this.parent.labels.range,
+        this.parent.labels.duration
+      ],
+      activation: () => {
+        const activation = firstActivity.activation ?? {};
+        return {
+          type: activation.type ?? "",
+          cost: activation.value ?? null,
+          condition: activation.condition ?? ""
+        };
+      },
+      consume: () => {
+        const consumption = firstActivity.consumption ?? {};
+        const target = consumption.targets?.[0] ?? {};
+        return {
+          type: target.type ?? "",
+          target: target.target ?? "",
+          amount: target.value ?? 1,
+          scale: consumption.scaling?.allowed ?? false
+        };
+      },
+      duration: () => firstActivity.duration ?? { value: null, units: "" },
+      hasAmmo: () => {
+        const consume = this.consume;
+        return this.isActive && !!consume.target && !!consume.type && this.hasAttack && (consume.type === "ammo");
+      },
+      hasAreaTarget: () => this.isActive && this.target.template?.type,
+      hasIndividualTarget: () => this.isActive && this.target.affects?.type,
+      hasResource: () => this.isActive && !!consume.target && !!consume.type && !this.hasAttack,
+      hasScalarDuration: () => this.duration.units in CONFIG.DND5E.scalarTimePeriods,
+      hasScalarRange: () => this.range.units in CONFIG.DND5E.movementUnits,
+      hasScalarTarget: () => this.target.template?.type || ![null, "", "self"].includes(this.target.affects?.type),
+      hasTarget: () => this.isActive && (this.target.template?.type || this.target.affects?.type),
```

Comment:
I guess these might still be useful if the Item has one of a `DurationField`, `RangeField`, or `TargetField`.

Reply:
We mainly have those for rendering the sheet changes which needs to be shared between item/activity sheets, so it would probably be best to re-implement these as necessary on the individual data fields instead.

Reply:
Yeah makes sense